xwiki-platform-core/xwiki-platform-ldap/xwiki-platform-ldap-authenticator/src/main/java/com/xpn/xwiki/plugin/ldap/XWikiLDAPConnection.java

```diff
@@ -375,11 +375,11 @@ protected void ldapToXWikiAttribute(List<XWikiLDAPSearchAttribute> searchAttribu
 
             LOGGER.debug("  - values for attribute [{}]", attributeName);
 
-            Enumeration<String> allValues = attribute.getStringValues();
+            Enumeration<byte[]> allValues = attribute.getByteValues();
 
             if (allValues != null) {
                 while (allValues.hasMoreElements()) {
-                    String value = allValues.nextElement();
+                    byte[] value = allValues.nextElement();
 
                     LOGGER.debug("    |- [{}]", value);
```

Comment:
You are creating Strings in XWikiLDAPSearchAttribute so you create the XWikiLDAPSearchAttribute before the log and print the String version of byte[] version depending on what XWikiLDAPSearchAttribute was able to do.